### DISCUSSION TITLE
Got “At line 172 of file H5_ff.F90 Fortran runtime error: Recursive call to nonrecursive procedure 'h5open_f'”

### DISCUSSION THREAD

**Discussion by cherry12306** [damask-multiphysics/DAMASK#40] [Q&A]
I have successfully installed the DAMASK 3.0.0 source code and completed the PETSc tests, all of which passed. Running some of the built-in examples works fine. However, when I attempt a nonlocal computation, I encounter the following error:

> At line 172 of file H5_ff.F90
Fortran runtime error: Recursive call to nonrecursive procedure 'h5open_f'

This issue occurred after reinstalling DAMASK; it was the same error I encountered during my first installation. It seems that HDF5 might not be handling parallel processing properly. Could you advise me on how to address this issue? I would greatly appreciate any assistance you could provide.

I downgraded gfortran to version 9.5. However, I encountered the following issue during computation:
>  
Increment 22566/1000000-11/16 @ Iteration 1≤56≤100

 deformation gradient aim       =
   1.0994363    0.0000000    0.0000000
   0.0045131    0.9395938    0.0000000
   0.0000000    0.0000000    1.0799794

 ... evaluating constitutive response ......................................
  Cell          489  failed (mechanics)

 Piola--Kirchhoff stress       / MPa =
       71.9801       489.7744         0.0000
      489.9044        24.7061        -0.0000
        0.0000        -0.0000        75.6616

 ... doing gamma convolution ...............................................

 ... evaluating constitutive response ......................................

 ┌─────────────────────────────────────────────────────────────────────┐
 │                        error 402                                    │
 ├─────────────────────────────────────────────────────────────────────┤
 │ invalid orientation specified                                       │
 │ fromMatrix                                                          │
 │                                                                     │
 └─────────────────────────────────────────────────────────────────────┘

 ┌─────────────────────────────────────────────────────────────────────┐
 │                        error 402                                    │
 ├─────────────────────────────────────────────────────────────────────┤
 │ invalid orientation specified                                       │
 │ fromMatrix                                                          │
 │                                                                     │
 └─────────────────────────────────────────────────────────────────────┘

 ┌─────────────────────────────────────────────────────────────────────┐
 │                        error 402                                    │
 ├─────────────────────────────────────────────────────────────────────┤
At line 172 of file H5_ff.F90
Fortran runtime error: Recursive call to nonrecursive procedure 'h5open_f'
 │ invalid orientation specified                                       │

Error termination. Backtrace:
 │ fromMatrix                                                          │
 │                                                                     │
 └─────────────────────────────────────────────────────────────────────┘
At line 172 of file H5_ff.F90
Fortran runtime error: Recursive call to nonrecursive procedure 'h5open_f'

Error termination. Backtrace:
 ┌─────────────────────────────────────────────────────────────────────┐
 │                        error 402                                    │
 ├─────────────────────────────────────────────────────────────────────┤
 │ invalid orient                                       │
 │ fromMatrix                                                          │
 │                                                                     │
 └─────────────────────────────────────────────────────────────────────┘
DAMASK_grid: H5SL.c:807: H5SL_insert_common: Assertion !_drop->forward[_i] || !(*(const hid_t *)((_drop->forward[_i])->key) < *(const hid_t *)key)' failed.
DAMASK_grid: H5SL.c:1250: H5SL_remove: Assertion !_drop->forward[_i] || !(*(const hid_t *)((_drop->forward[_i])->key) < *(const hid_t *)key)' failed.

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
#0  0x7f85f5423e59 in ???
#1  0x7f85f5424a71 in ???
#2  0x7f85f5425082 in ???
#3  0x7f85f7837ae8 in __h5lib_MOD_h5open_f
	at /home/cairui/DAMASK/petsc-3.15.3/arch-linux-c-opt/externalpackages/hdf5-1.12.0/fortran/src/H5_ff.F90:172
#4  0x5c67a0b724d7 in ???
#5  0x5c67a0aa8339 in ???
#6  0x5c67a0b7a31e in ???
#7  0x5c67a0ae79a7 in ???
#8  0x5c67a0ac891a in ???
#9  0x7f85f5cbe976 in ???
#10  0x5c67a0ac9bad in ???
#11  0x5c67a0bb7a1e in ???
#12  0x5c67a0ba23d0 in ???
#13  0x7f85f6bfb634 in ???
#14  0x7f85f6bdeed9 in ???
#15  0x7f85f6c0f199 in ???
#16  0x7f85f6ba8630 in ???
#17  0x7f85f6ba678d in ???
#18  0x7f85f6c1d268 in ???
#19  0x7f85f6c26ed5 in ???
#20  0x5c67a0ba5555 in ???
#21  0x5c67a0b89305 in ???
#22  0x5c67a0a9710e in ???
#23  0x7f85f502a1c9 in __libc_start_call_main
	at ../sysdeps/nptl/libc_start_call_main.h:58
#24  0x7f85f502a28a in __libc_start_main_impl
	at ../csu/libc-start.c:360
#25  0x5c67a0a97144 in ???
#26  0xffffffffffffffff in ???
DAMASK_grid: H5SL.c:1250: H5SL_remove: Assertion !_drop->forward[_i] || !(*(const hid_t *)((_drop->forward[_i])->key) < *(const hid_t *)key)' failed.

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
#0  0x7f85f5423e59 in ???
#1  0x7f85f5424a71 in ???
#2  0x7f85f5425082 in ???
#3  0x7f85f7837ae8 in __h5lib_MOD_h5open_f
	at /home/cairui/DAMASK/petsc-3.15.3/arch-linux-c-opt/externalpackages/hdf5-1.12.0/fortran/src/H5_ff.F90:172
#4  0x5c67a0b724d7 in ???
#5  0x5c67a0aa8339 in ???
#6  0x5c67a0b7a31e in ???
#7  0x5c67a0ae79a7 in ???
#8  0x5c67a0ac891a in ???
#9  0x7f85f5cc879e in ???
#10  0x7f85f509ca93 in start_thread
	at ./nptl/pthread_create.c:447
#11  0x7f85f5129c3b in clone3
	at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:78
#12  0xffffffffffffffff in ???

I’d like to ask if you've encountered a similar issue before. This seems to be an HDF5 call error.

**Comment by MarDiehl** [comment]
pls. don't double post.
